### ISSUE TITLE
`AttributeError: 'Lookahead' object has no attribute '_optimizer_step_pre_hooks'` in Pytorch 2.0

### ISSUE THREAD

**Issue by kozistr** [kozistr/pytorch_optimizer#122]
In Pytorch 2.0, the optimizer needs to have `optimizer_step_pre_hooks` method in the class, but, Lookahead doesn't.

**Comment by FPX-Rain** [comment]
Do you know how to solution this problem now? If you know , could you tell me how to solution this problem ??

**Comment by kozistr** [comment]
> Do you know how to solution this problem now? If you know , could you tell me how to solution this problem ??

Sure! From Pytorch 2.0, the class that inherits `torch.optim.Optimizer` needs to call the super class with parameters to init the optimizer class (like `super().__init__(params, defaults)`). However, many Lookahead implementations in Github don't call the super class while inheriting the `torch.optim.Optimizer` class.

In short, I simply made it not inherit `torch.optim.Optimizer` class. (Lookahead is just a wrapper class, so, actually, doesn't really need to inherit it).

Also, you can check my version of the implementation [here](https://github.com/kozistr/pytorch_optimizer/blob/main/pytorch_optimizer/optimizer/lookahead.py).

I hope it helps you :)

thank you!

**Comment by FPX-Rain** [comment]
Your answer has been very helpful to me
Thank you for your kind answer




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Hyeongchan ***@***.***&gt;; 
发送时间: 2023年6月5日(星期一) 下午5:24
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [kozistr/pytorch_optimizer] `AttributeError: &#39;Lookahead&#39; object has no attribute &#39;_optimizer_step_pre_hooks&#39;` in Pytorch 2.0 (Issue #122)





  
Do you know how to solution this problem now? If you know , could you tell me how to solution this problem ??
  
Sure! From Pytorch 2.0, the class that inherits torch.optim.Optimizer needs to call the super class with parameters to init the optimizer class (like super().__init__(params, defaults)). However, many Lookahead implementations in Github don't call the super class while inheriting the torch.optim.Optimizer class.
 
In short, I simply made it not inherit torch.optim.Optimizer class. (Lookahead is just a wrapper class, so, actually, doesn't really need to inherit it).
 
Also, you can check my version of the implementation here.
 
I hope it helps you :)
 
thank you!
